tests/unit/test_xcelium_interface.py

```diff
@@ -0,0 +1,1150 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# Copyright (c) 2014-2021, Lars Asplund lars.anders.asplund@gmail.com
+
+# pylint: disable=too-many-public-methods, too-many-lines
+
+"""
+Test the Xcelium interface
+"""
+
+
+import unittest
+from pathlib import Path
+import os
+from shutil import rmtree
+from unittest import mock
+from vunit.sim_if.xcelium import XceliumInterface
+from vunit.project import Project
+from vunit.ostools import renew_path, write_file, read_file
+from vunit.test.bench import Configuration
+from vunit.vhdl_standard import VHDL
+
+
+class TestXceliumInterface(unittest.TestCase):
+    """
+    Test the Xcelium interface
+    """
+
+    @mock.patch("vunit.sim_if.xcelium.XceliumInterface.find_cds_root_virtuoso")
```

Comment:
Similar mocks are used for most test methods. I would apply all the common ones to the class. For any duplicated code, you can set up a static or class method to configure the default mocks or leave customized mocks local to the test methods if it's not shared.